### PR TITLE
feat(Videoland): add presence

### LINKED ISSUE #5459: videoland | www.videoland.com
### Discussed in https://github.com/PreMiD/Presences/discussions/4464

<div type='discussions-op-text'>

<sup>Originally posted by **ChihabTemsamani** May  2, 2020</sup>
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [x] Is the site locked to a specific country/region?
i think only in the netherlands
- [x] Is the site a paid service? (e.g. Netflix, Hulu)
- [x] Does the website feature NSFW content? (e.g. porn, etc...)
in some movies and series 
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?**  ! Chihab Temsamani#1984

**What's the name of the service?**  videoland

**What should the Presence display?**  what movie or serie is the user watching 

**If possible, please provide a logo for the service (512x512 minimum)**  https://prnt.sc/s9h40b
</div>

### PR DESCRIPTION
**Pull request by DarkVillager**
Resolves #5459 

**Describe the changes in this pull request:**
- Added Videoland presence 

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

</details>

(Btw thanks @Slowlife01 me stupid sometimes)

### CODE REVIEW

File: websites/V/Videoland/presence.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp
+		},
+		page = window.location.pathname;
+
+	if (page === "/") presenceData.details = "Bekijkt de homepagina";
+	else if (page.includes("zoeken")) {
+		search = document.querySelector("#search");
```
Comment (EncryptedDev) [inline websites/V/Videoland/presence.ts:16]: Search is never redefined, make this a constant instead

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/V/Videoland/presence.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			"meta[property='og:title']"
+		).content;
+	} else if (page.includes("/player/")) {
+		delete presenceData.startTimestamp;
+		const f = await fetch(
+			`https://www.videoland.com/api/v3/titles/${
+				window.location.href.split("/")[4]
+			}/details`
+		).then(x => x.json());
+		presenceData.details = await f.showTitle;
```
Comment (Bas950) [inline websites/V/Videoland/presence.ts:37]: yea maybe dont do this in updatedata... 
and cache/save previously fetched ones

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/V/Videoland/presence.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+				.replace(f.showTitle, "S")
+				.replace(".", ":E")
+				.replace(" ", "");
+		}
+		if (document.querySelector("video").paused) {
+			delete presenceData.endTimestamp;
+			presenceData.smallImageKey = "pause";
+			presenceData.smallImageText = "Gepauzeerd";
+		} else {
+			presenceData.endTimestamp = presence.getTimestamps(
```
Comment (Slowlife01) [inline websites/V/Videoland/presence.ts:56]: Use `getTimestampsfromMedia`

File: websites/V/Videoland/presence.ts (hunk 2 above)
Comment (Slowlife01) [reply websites/V/Videoland/presence.ts:37]: also more appropriate variable name

**Comment by DarkVillager** [comment]
currently don’t have a pc bruh

**Comment by DarkVillager** [comment]
<details>
<summary> Updated proof showing the creation/modification is working as expected </summary>
<br>






</details>

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**